test/framework/e2e.go

```diff
@@ -107,6 +108,10 @@ func (e *E2ETest) CreateCluster() {
 	e.cleanup(func() {
 		os.RemoveAll(e.ClusterName)
 	})
+	// Setting GitRepo cleanup if the test has GitOps configured
+	if e.GitOpsConfig != nil {
+		e.T.Cleanup(e.CleanUpGithubRepo)
```

Comment:
You could do this from `WithFlux`, it would be cleaner, no need to check for `e.GitOpsConfig != nil`